test/linalg/test_linalg_qr.fypp

```diff
@@ -0,0 +1,141 @@
+#:include "common.fypp"
+#:set RC_KINDS_TYPES = REAL_KINDS_TYPES + CMPLX_KINDS_TYPES
+! Test QR factorization 
+module test_linalg_qr
+    use testdrive, only: error_type, check, new_unittest, unittest_type
+    use stdlib_linalg_constants
+    use stdlib_linalg_state, only: LINALG_VALUE_ERROR,linalg_state_type
+    use stdlib_linalg, only: qr,qr_space
+
+    implicit none (type,external)
+    
+    public :: test_qr_factorization
+
+    contains
+
+    !> QR factorization tests
+    subroutine test_qr_factorization(tests)
+        !> Collection of tests
+        type(unittest_type), allocatable, intent(out) :: tests(:)
+        
+        allocate(tests(0))
+        
+        #:for rk,rt,ri in RC_KINDS_TYPES
+        tests = [tests,new_unittest("qr_random_${ri}$",test_qr_random_${ri}$)]
+        #:endfor
+
+    end subroutine test_qr_factorization
+
+    !> QR factorization of a random matrix
+    #:for rk,rt,ri in RC_KINDS_TYPES
+    subroutine test_qr_random_${ri}$(error)
+        type(error_type), allocatable, intent(out) :: error
+
+        integer(ilp), parameter :: m   = 15_ilp
+        integer(ilp), parameter :: n   =  4_ilp
+        integer(ilp), parameter :: k   = min(m,n)
+        real(${rk}$), parameter :: tol = 100*sqrt(epsilon(0.0_${rk}$))
+        ${rt}$ :: a(m,n),aorig(m,n),q(m,m),r(m,n),qred(m,k),rred(k,n),qerr(m,6),rerr(6,n)
+        real(${rk}$) :: rea(m,n),ima(m,n)
+        integer(ilp) :: lwork
+        ${rt}$, allocatable :: work(:)
+        type(linalg_state_type) :: state
+        
+        call random_number(rea)
+        #:if rt.startswith('complex')
+        call random_number(ima)
+        a = cmplx(rea,ima,kind=${rk}$)
+        #:else
+        a = rea
+        #:endif
+        aorig = a
+        
+        ! 1) QR factorization with full matrices. Input NaNs to be sure Q and R are OK on return
+        q = 0.0_${rk}$
+        q = 1.0_${rk}$/q
+        r = 0.0_${rk}$
+        r = 1.0_${rk}$/r       
```

Comment:
Why not use the ieee module?
```suggestion
        q = ieee_value(0.0_${rk}$,  ieee_quiet_nan)
        r = ieee_value(0.0_${rk}$,  ieee_quiet_nan)     
```

Reply:
[6fc0e1d](https://github.com/fortran-lang/stdlib/pull/832/commits/6fc0e1d87465723abb499dbdd58b034379a793a2)